### PR TITLE
Add ability to post pip install after project installation

### PR DESCRIPTION
**Pull request by edtubillara**
This adds the ability to install any dependencies which need
to get pip installed after the project gets installed. This
is done for both source and distro install.

### CODE REVIEW

File: roles/keystone-defaults/defaults/main.yml [hunk 1]
```diff
@@ -24,6 +24,7 @@ keystone:
       - { name: uwsgi }
       - { name: pyldap }
       - { name: ldappool }
+    python_post_dependencies: []
```
Comment (nirajdp76) [inline roles/keystone-defaults/defaults/main.yml]: minor nit, can u rename this to python_post_deps: []

File: roles/keystone/meta/main.yml [hunk 2]
```diff
@@ -14,15 +14,17 @@ dependencies:
     alternatives: "{{ keystone.alternatives }}"
     system_dependencies: "{{ keystone.source.system_dependencies }}"
     python_dependencies: "{{ keystone.source.python_dependencies }}"
+    python_post_dependencies: "{{ keystone.source.python_post_dependencies }}"
```
Comment (nirajdp76) [inline roles/keystone/meta/main.yml]: same variable name change comment as above.

File: roles/openstack-distro/tasks/osp.yml [hunk 3]
```diff
@@ -16,3 +16,13 @@
   retries: 5
   with_items: "{{ dependent_packages }}"
   when: dependent_packages is defined
+
+- name: python post dependencies for project
+  pip:
+    name: "{{ item.name }}"
+    version: "{{ item.version|default(omit) }}"
+    extra_args: "{{ item.pip_extra_args|default(omit) }}"
+  with_items: "{{ openstack_source.python_post_deps }}"
```
Comment (scoop206) [inline roles/openstack-distro/tasks/osp.yml]: should be `openstack_distro.python_post_deps` ?

Comment (nirajdp76) [inline roles/openstack-distro/tasks/osp.yml]: Need to also port global pip_extra_args logic from openstack-source to openstack-distro role.

**Comment by nirajdp76** [comment]
ok to test

**Pull request closed without merging**